File: src/main/java/burrows/apps/example/gif/presentation/main/IMainView.java
package burrows.apps.example.gif.presentation.main;

import burrows.apps.example.gif.data.rest.model.RiffsyResponse;
import burrows.apps.example.gif.presentation.IBaseView;
import burrows.apps.example.gif.presentation.adapter.model.ImageInfoModel;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
interface IMainView extends IBaseView<IMainPresenter> {
  void clearImages();
  void addImages(RiffsyResponse response);
  void showDialog(ImageInfoModel url);
  boolean isActive();
}

File: src/main/kotlin/burrows/apps/example/gif/presentation/main/IMainView.kt
package burrows.apps.example.gif.presentation.main

import burrows.apps.example.gif.data.rest.model.RiffsyResponseDto
import burrows.apps.example.gif.presentation.IBaseView
import burrows.apps.example.gif.presentation.adapter.model.ImageInfoModel

/**
 * @author [Jared Burrows](mailto:jaredsburrows@gmail.com)
 */
interface IMainView : IBaseView<IMainPresenter> {
  fun clearImages()
  fun addImages(response: RiffsyResponseDto)
  fun showDialog(imageInfoModel: ImageInfoModel)
  fun isActive(): Boolean
}
